### PR TITLE
React: Add render component for multiple files on Datasets page

### LINKED ISSUE #280: Datasets page: show multiple files like add datasets page
Spun off from #230

### PR DESCRIPTION
**Pull request by frewmack**
Closes #280

### CODE REVIEW

File: react/src/Admin/components/UserList.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ function reducer(state: User[], action: AddUserAction | ChangeUserAction | SetUs
             const newUser = action.payload;
             return state.concat({
                 username: newUser.username,
-                email: newUser.username,
+                email: newUser.email,
```
Comment (kevinlul) [inline react/src/Admin/components/UserList.tsx:73]: oops

File: react/src/Datasets/components/LinkedFilesPopover.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        fileNames: string[];
+    } & PopoverProps
+) {
+    const classes = useStyles();
+    const { fileNames, ...popoverProps } = props;
+
+    return (
+        <Popover {...popoverProps}>
+            {props.fileNames.map(name => (
+                <Typography className={classes.typography}>{name}</Typography>
```
Comment (kevinlul) [inline react/src/Datasets/components/LinkedFilesPopover.tsx:21]: probably needs `key=`

File: react/src/Datasets/components/LinkedFilesButton.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    vertical: "bottom",
+                    horizontal: "center",
+                }}
+                transformOrigin={{
+                    vertical: "top",
+                    horizontal: "center",
+                }}
+            />
+            <ButtonBase disabled={length === 0} onClick={handleClick} className={classes.button}>
+                <span>{`${length} file${length === 1 ? "" : "s"}`}</span>
```
Comment (kevinlul) [inline react/src/Datasets/components/LinkedFilesButton.tsx:44]: 👍

File: react/src/Datasets/components/DatasetTable.tsx [hunk 4]
```diff
@@ -155,9 +156,14 @@ export default function DatasetTable({ isAdmin }: DatasetTableProps) {
...
                     {
-                        title: "File",
-                        field: "input_hpf_path",
+                        title: "Files",
+                        field: "linked_files",
                         grouping: false,
+                        // can search by number of files, or by file name
+                        customFilterAndSearch: (filter: string, rowData) =>
+                            filter === "" + rowData.linked_files.length ||
+                            rowData.linked_files.some(name => name.includes(filter)),
```
Comment (kevinlul) [inline react/src/Datasets/components/DatasetTable.tsx:165]: I like this idea!

**Review by kevinlul — CHANGES_REQUESTED** [review]

**Review by kevinlul — CHANGES_REQUESTED** [review]
Seems good. I assume selecting and rendering multiple files in the dropdown is beyond the scope of this?

**Comment by frewmack** [comment]
> 
> 
> Seems good. I assume selecting and rendering multiple files in the dropdown is beyond the scope of this?

Yes editing the linked files in the table is beyond the scope of this particular issue.

**Review by kevinlul — APPROVED** [review]

File: react/src/Datasets/components/DatasetTable.tsx [hunk 5]
```diff
@@ -155,9 +156,14 @@ export default function DatasetTable({ isAdmin }: DatasetTableProps) {
...
+                        render: data => <LinkedFilesButton fileNames={data.linked_files} />,
```
Comment (annnag) [inline react/src/Datasets/components/DatasetTable.tsx:166]: Probably need to add `customSort` here to make sorting work.

**Review by annnag — CHANGES_REQUESTED** [review]
Looks good! One small problem that I noticed is sorting doesn't work properly on the files column.

**Pull request merged**